src/Common/Auth/AuthUtils.php

```diff
@@ -967,4 +961,55 @@ private function clearFromMemory(&$password)
             $password = '';
         }
     }
+
+    /**
+     * Validates a google ID token and returns a user on success. If validation
+     * fails, return false.
+     *
+     * @param $token
+     * @return array|bool
+     */
+    static public function verifyGoogleSignIn($token)
+    {
```

Comment:
Recommend methodically going through the items (akin as is done in confirmUserPassword function) for the following (if fail at a step, then log it and return false; also, if succeed, then log the success; in logging make clear using the google sign in mechanism):
1. Check to ensure user exists and is active
2. Check to ensure user has a google sign in email address
3. Check to ensure user is in a group (and collect the group name)
4. Check to ensure user is in a acl group
5. Check authentication (get payload with token)
6. Set up user session variables

Reply:
And then this function will simply return true or false

Reply:
OK.  Working on this now and I should have the PR in a few

Reply:
OK.  Mostly following these directions.  If the user enters a google email that doesn't exist or is not part of the GSuite existing email addresses, there is no signin attempt from OpenEMR standpoint.  If I try to log in from an account other than from mi-squared domain or if the email doesn't exist, OpenEMR never sees the request.  The disadvantage to that is we can't log these attempts from OpenEMR's end. 

 getUserByGoogleSignIn checks if 1) there is a Gsuite email address associated with that user and it has been added to the user table, and 2) the user must be active.

Reply:
@growlingflea Ii don't think Brady is talking about the authentication on Google's end, just make sure everything we check is logged, and we're missing the check for "Active" and "ACL" group. 

    Check to ensure user exists and is active

Make sure the OpenEMR user associated with the Google JWT exists and is active. If not, log it. The code for this is in AuthUtils::confirmUserPassword() and could be moved to it's own function called verifyUserExists() since it will be shared.

    Check to ensure user is in a group (and collect the group name for logging purposes)

Ensure the OpenEMR user is in a group. This is currently done in auth.inc. Move it into the verifyGoogleSignin() function.

    Check to ensure user has a google sign in email address

This is already done above in first check.

    Check to ensure user is in a acl group

Code for this is also in AuthUtils::confirmUserPassword()

    Check authentication

 we do this, just need logging if failure.

    Set up user session variables

We do this.

Reply:
@kchapple 

I interpreted Brady's comment as moving all the logging into the verifyGoogleSignin function.  I'll post the changes I made soon.  I have a few errands to run and I should be back in a few hours.